openff/toolkit/utils/_nagl_wrapper.py

```diff
@@ -98,9 +98,9 @@ def assign_partial_charges(
                 raise FileNotFoundError(f"Could not find model {_only_model.name}")
 
             model = GNNModel.load(_only_model, eval_mode=True)
-
+            charges = model.compute_property(molecule, as_numpy=True)
             molecule.partial_charges = unit.Quantity(
-                model.compute_property(molecule, as_numpy=True),
+                charges.astype(float),
```

Comment:
I was curious how memory inefficient this is, turns out even more trivial than I thought

```python
In [1]: from openff.toolkit.utils._nagl_wrapper import _NAGLToolkitWrapper

In [2]: from openff.toolkit import Molecule

In [3]: from rdkit import Chem

In [4]: molecule = Molecule.from_rdkit(Chem.MolFromSequence("YGGFMGGY"))

In [5]: molecule.assign_partial_charges(
   ...:     partial_charge_method="_nagl_am1bccelf10",
   ...:     toolkit_registry=_NAGLToolkitWrapper(),
   ...: )
/Users/mattthompson/mambaforge/envs/openff-toolkit-test/lib/python3.10/site-packages/dgl/backend/pytorch/tensor.py:449: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
  assert input.numel() == input.storage().size(), (

In [6]: molecule.partial_charges.m.nbytes
Out[6]: 440

In [7]: molecule.partial_charges.m.astype(float).nbytes
Out[7]: 880

In [8]: molecule.n_atoms
Out[8]: 110